sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -611,6 +613,12 @@ public void init(CosmosClientMetadataCachesSnapshot metadataCachesSnapshot, Func
                 this.initializeDirectConnectivity();
             }
             this.retryPolicy.setRxCollectionCache(this.collectionCache);
+            ConsistencyLevel effectiveConsistencyLevel = consistencyLevel != null
```

Comment:
So, session token capturing in `SessionContainer` is falling back to client-side settings. Wouldn't this lead to session token being captured in case of account-level consistencies weaker than Session consistency? I assume customers doing account-level changes (changing consistency from Eventual to Session) is less prevalent than client-level changes.

Reply:
Capturing session tokens in that case is fine. Whether Session consistency is used depends on two things - effective consistency level for a request and whether session token is present (customer provided or in the session token container). Setting client level consistency to Session and account level consistency being eventual was possible before a swell.

Reply:
In general Cient-setting overrides account default
RequestOptions override client-setting

Reply:
Downgrades (requested consistency less than account default is only applicable for reads/queries)

Reply:
Upgrades should always be ignored